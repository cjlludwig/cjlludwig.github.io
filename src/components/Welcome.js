import Typewriter from 'typewriter-effect';
import '../static/styles/components/Welcome.css';

function Welcome() {
  return (
    <div className='typewriter-wrapper'>
      <Typewriter
        onInit={(typewriter) => {
          typewriter.typeString('Hello!')
            .pauseFor(300)
            .deleteAll()
            .typeString("\nMy name's Connor.<br>")
            .pauseFor(300)
            .typeString("\nWelcome to my site!<br>")
            .pauseFor(2000)
            .typeString("\nScroll on for more...")
            .start();
        }}
      />
    </div>
  );
}

export default Welcome;

