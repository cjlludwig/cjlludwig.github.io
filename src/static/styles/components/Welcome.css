.typewriter-wrapper {
  font-family: monospace;
  padding-top: 35vh;
  padding-left: 25%;
  padding-right: 15%;
  font-size: 3rem;
  color: white;
}

/* Mobile */
@media screen and (max-width:960px) {
  /* CSS for screens that are 480px pixels or less will be put in this section */
  .typewriter-wrapper {
    padding-top: 35vh;
    padding-left: 15%;
    font-size: 2.5rem;
    color: white;
  }

}