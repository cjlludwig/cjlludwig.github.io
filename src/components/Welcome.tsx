import Typewriter from 'typewriter-effect';
import '../static/styles/components/Welcome.css';

function Welcome() {
  return (
    <div className='typewriter-wrapper'>
      <Typewriter
        onInit={(typewriter) => {
          typewriter
            .typeString('Hi, I\'m Connor Ludwig')
            .pauseFor(1000)
            .deleteAll()
            .typeString('Staff Software Engineer @ Built')
            .pauseFor(1000)
            .deleteAll() 
            .typeString('Scroll to explore')
            .start();
        }}
        options={{
          delay: 50,
        }}
      />
    </div>
  );
}

export default Welcome;
