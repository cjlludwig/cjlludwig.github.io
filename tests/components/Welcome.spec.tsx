import React from "react";
import renderer from "react-test-renderer";

import Welcome from "../../src/components/Welcome";
import Typewriter from 'typewriter-effect';

jest.mock("typewriter-effect");

describe("<Welcome />", () => {
  test("renders correctly", () => {
    const tree = renderer
      .create(<Welcome />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});