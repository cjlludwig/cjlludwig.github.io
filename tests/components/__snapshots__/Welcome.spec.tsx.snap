// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Welcome /> renders correctly 1`] = `
<div
  className="typewriter-wrapper"
/>
`;
